cirq/ops/gate_operation.py

```diff
@@ -186,13 +186,19 @@ def _qasm_(self, args: protocols.QasmArgs) -> Optional[str]:
                               qubits=self.qubits,
                               default=None)
 
+
+class IdentityOperation(GateOperation):
```

Comment:
Inherit from `Operation`, not `GateOperation`.

Reply:
Rule of thumb: whenever you find yourself throwing exceptions like "I know I said I'm an X but I don't do the thing Y that Xs normally do", you should not be declaring yourself to be an X.

Reply:
Again, fair criticism, thanks. I was just surprised how much duplicate code from GateOperation I had to introduce in the IdentityOperation when inheriting from Operation to make it past the older tests. Methods like `__init__`, `__repr__`, `__str__`, `_pauli_expansion_` and more are exactly the same. That didn't look good to me either.

Let me try again and let you be the judge.